src/NLog/Layouts/Layout.cs

```diff
@@ -271,35 +268,45 @@ void ISupportsInitialize.Close()
         /// Initializes this instance.
         /// </summary>
         /// <param name="configuration">The configuration.</param>
-		internal void Initialize(LoggingConfiguration configuration)
-		{
+        internal void Initialize(LoggingConfiguration configuration)
+        {
             if (!this.isInitialized)
             {
                 this.LoggingConfiguration = configuration;
                 this.isInitialized = true;
+                this.scannedForObjects = false;
+
+                this.InitializeLayout();
```

Comment:
For the normalt NLog Layouts, then the scan is purely done for sake of the Target and the later processing of LogEventInfos. Pretty sure no one knows how threadagnostic and stacktraceusage works :)